oop1-construct.js

```diff
@@ -20,58 +66,99 @@ Create a class that can categorize at least 6 of the creatures that you recall o
 
 class Shape {
     //your code here...
+    constructor(name, sides, base, height, area, length, width) {
+        this.name = name;
+        this.sides = sides;
+        this.base = base;
+        this.height = height;
+        this.area = area;
+        this.length = length;
+        this.width = width;
+    }
+
+
 }
+class Triangle extends Shape {
+    constructor(name, sides, base, height, area) {
+        super(name, sides, base, height, area);
+    }
+
 
+    HeronsCalc() {
+        let heronsStart = this.sides.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
+        let variableS = heronsStart / 2;
+        let triangleArea = Math.sqrt(variableS * ((variableS - this.sides[0]) * (variableS - this.sides[1]) * (variableS - this.sides[2])));
+        let triangleHeight = 2 * (triangleArea / this.sides[0]);
+        this.height = triangleHeight;
+        this.base = this.sides[0];
+        this.area = triangleArea;
 
-const triangle = {
-    name : "triangle",
-    sides : [4,7,7], //lengths of each side
-    base : null,
-    height : null, //determine this height using basic geometry.  You may need to research this one.
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${.5 * this.base * this.height}`);
-    },
-    calcPerimeter : function() {
-        console.log(`${this.name}'s perimeter is calculated to be : ${(this.base) + (this.sides[1] + this.sides[2]) }`)
     }
 }
 
-triangle.base = triangle.sides[0];  //base is the 1st side in the triangle.sides array.
-triangle.height = 6.71; //determine the height using basic geometry.  How do you calculate the height of a triangle with 2 equal sides?  If you use the formula to caluclate the height, this is a bonus.  If you hard code the correct value here, that will be sufficient, but no bonus :(
+triangle = new Triangle("triangle", [4, 7, 7], null, null);
 
-console.log(triangle);
-triangle.calcArea();
-triangle.calcPerimeter();
+triangle.HeronsCalc();
 
-const rectangle = {
-    name : "rectangle",
-    sides : 4,
-    length : 2,
-    width : 5,
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${this.length * this.width}`);
-    },
-    calcPerimeter : function() {
-        console.log(`${this.name}'s perimeter is calculated to be : ${(2 * this.length) + (2 * this.width) }`)
+
+shape = new Shape("triangle", [4, 7, 7], 8, 8);
```

Comment:
What is this line in for? You never use `shape` anywhere. If you don't use it anywhere, take it out.